src/mono/mono/mini/interp/transform.c

```diff
@@ -3435,7 +3435,10 @@ interp_emit_swiftcall_struct_lowering (TransformData *td, MonoMethodSignature *c
 				}
 			} else {
 				// For structs that cannot be lowered, we change the argument to byref type
-				ptype = mono_class_get_byref_type (mono_defaults.typed_reference_class);
+				if ((strcmp (klass->name, "SwiftSelf`1") == 0) && (strcmp (klass->name_space, "System.Runtime.InteropServices.Swift") == 0))
+					ptype = mono_class_get_byref_type (swift_self);
+				else
+					ptype = mono_class_get_byref_type (mono_defaults.typed_reference_class);
```

Comment:
I know this bit isn't new, but did we add TypedReference as a dynamic dependency to `SwiftSelf` / `SwiftSelf<T>` for the linker?

Reply:
I don't think so. Would the build fail if it's not added as a dynamic dependency?